carma-messenger-core/cpp_message/src/Map_Message.cpp

```diff
@@ -0,0 +1,627 @@
+                    rclass.id = rca->id;
+                    output.restriction_list.restriction_class_list.push_back(rclass);
+                }
+            }//end Restriction List
+            else
+            {
+                output.restriction_list_exists = false;
+            }
+
+
+            //Road Segment
+            if(map_msg.roadSegments)
+            {
+
+                output.road_segments_exists = true;
+
+                RoadSegment_t *rseg = new RoadSegment_t;
+
+                for(size_t i = 0; i< map_msg.roadSegments->list.size;i++)
+                {
+                    rseg = map_msg.roadSegments->list.array[i];
+
+                    j2735_msgs::RoadSegment rs;
+
+                    if(rseg->laneWidth)
+                    {
+                        rs.lane_width_exists = true;
+                        rs.lane_width = *rseg->laneWidth;
+                    }
+                    else
+                    {
+                        rs.lane_width_exists = false;
+                    }
+
+                    //Road Segment Name
+                    if(rseg->name)
+                    {
+                        rs.name_exists = true;
+                        
+                        DescriptiveName_t *nm = new DescriptiveName_t;
+
+                        nm = rseg->name;
+                        std::string n;
+                        
+                        for (size_t i = 0; i < nm->size;i++)
+                        {
+                            n.push_back(nm->buf[i]);
+                        }
+                        rs.name = n;
+                    }
+                    else
+                    {
+                        rs.name_exists = false;
+                    }
+
+                    if (rseg->laneWidth)
+                    {
+                        rs.lane_width_exists = true;
+
+                        rs.lane_width = *rseg->laneWidth;
+                    }
+                    else
+                    {
+                        rs.lane_width_exists = false;
+                    }
+
+                    if(rseg->speedLimits)
+                    {
+                        rs.speed_limits_exists = true;
+
+                        RegulatorySpeedLimit_t *speedL = new RegulatorySpeedLimit_t;
+
+                        for (size_t i = 0; i < rseg->speedLimits->list.size;i++)
+                        {
+                            j2735_msgs::RegulatorySpeedLimit sl;
+
+                            speedL = rseg->speedLimits->list.array[i];
+
+                            sl.speed = speedL->speed;
+                            sl.type.speed_limit_type = speedL->type;
+                            rs.speed_limits.speed_limits.push_back(sl);
+                        }
+                    }//end speed limits
+                    else
+                    {
+                        rs.speed_limits_exists = false;
+                    }
+
+                    if(rseg->refPoint.elevation)
```

Comment:
Same not as previous Position3d, should elevation definition be used to check for lat/lon as well?